### PR TITLE
 #4913 Add payment discount when importing discount schema

### PR DESCRIPTION
**Pull request by cristinamghita**
#4913 Add payment discount when importing discount schema

### CODE REVIEW

File: de.metas.business/src/main/java/de/metas/bpartner/impexp/BPartnerImportTableSqlUpdater.java [hunk 1]
```diff
@@ -640,7 +640,7 @@ private void dbUpdateM_PricingSystems(final String whereClause)
 		int no;
 		sql = new StringBuilder("UPDATE I_BPartner i "
 				+ "SET M_PricingSystem_ID=(SELECT M_PricingSystem_ID FROM M_PricingSystem ps"
-				+ " WHERE i.PricingSystem_Value=ps.value AND ps.AD_Client_ID IN (0, i.AD_Client_ID)) "
+				+ " WHERE i.PricingSystem_Value=ps.value AND ps.AD_Client_ID IN (0, i.AD_Client_ID) and IsActive='Y' ) "
```
Comment (metas-ts) [inline de.metas.business/src/main/java/de/metas/bpartner/impexp/BPartnerImportTableSqlUpdater.java]: does this really work? after all, both `i` and `ps` have an `IsActive` column..i would guess

```suggestion
				+ " WHERE i.PricingSystem_Value=ps.value AND ps.AD_Client_ID IN (0, i.AD_Client_ID) and ps.IsActive='Y' ) "
```

File: de.metas.business/src/main/java/de/metas/bpartner/impexp/BPartnerImportTableSqlUpdater.java [hunk 2]
```diff
@@ -660,7 +660,7 @@ private void dbUpdatePO_PricingSystems(final String whereClause)
 		int no;
 		sql = new StringBuilder("UPDATE I_BPartner i "
 				+ "SET M_PricingSystem_ID=(SELECT M_PricingSystem_ID FROM M_PricingSystem ps"
-				+ " WHERE i.PO_PricingSystem_Value=ps.value AND ps.AD_Client_ID IN (0, i.AD_Client_ID)) "
+				+ " WHERE i.PO_PricingSystem_Value=ps.value AND ps.AD_Client_ID IN (0, i.AD_Client_ID) and IsActive='Y') "
```
Comment (metas-ts) [inline de.metas.business/src/main/java/de/metas/bpartner/impexp/BPartnerImportTableSqlUpdater.java]:
```suggestion
				+ " WHERE i.PO_PricingSystem_Value=ps.value AND ps.AD_Client_ID IN (0, i.AD_Client_ID) and ps.IsActive='Y') "
```

File: de.metas.business/src/main/java/de/metas/pricing/impexp/MDiscountSchemaImportTableSqlUpdater.java [hunk 3]
```diff
@@ -119,6 +120,26 @@ private void dbUpdateM_PricingSystems(final String whereClause)
...
+				.append("AND dsb.c_paymentterm_id = i.c_paymentterm_id ")
+				.append("AND coalesce(dsb.breakdiscount, 0) = coalesce(i.breakdiscount,0) ")
+				.append("AND coalesce(dsb.breakvalue, 0) = coalesce(i.breakvalue,0) ")
+				.append("AND dsb.PriceBase = i.PriceBase ")
+				.append("AND dsb.Base_PricingSystem_ID = i.Base_PricingSystem_ID ")
+				.append("AND coalesce(dsb.PriceStdFixed, 0) = coalesce(i.PriceStdFixed, 0) ")
+				.append("AND coalesce(dsb.PricingSystemSurchargeAmt, 0) = coalesce(i.PricingSystemSurchargeAmt,0) ")
+				.append("AND coalesce(dsb.C_Currency_ID,0) = coalesce(i.C_Currency_ID,0) ")
+				.append("AND i.M_DiscountSchemaBreak_ID IS NULL ")
+				.append("AND I_IsImported<>'Y' ");
```
Comment (metas-ts) [inline de.metas.business/src/main/java/de/metas/pricing/impexp/MDiscountSchemaImportTableSqlUpdater.java]:
```suggestion
				.append("AND I_IsImported <> 'Y' ")
				.append("AND dsb.IsActive = 'Y' ");
```

File: de.metas.business/src/main/java/de/metas/bpartner/impexp/BPartnerImportTableSqlUpdater.java (hunk 1 above)
Comment (cristinamghita) [reply de.metas.business/src/main/java/de/metas/bpartner/impexp/BPartnerImportTableSqlUpdater.java]: Yes; weird,  but is working..

**Pull request merged**